tockloader-lib/src/tabs/tab.rs

```diff
@@ -75,6 +76,52 @@ impl Tab {
         }
     }
 
+    pub fn filter_tbfs(
+        &self,
+        settings: &BoardSettings,
+    ) -> Result<Vec<(&TbfFile, u64, u64)>, TockloaderError> {
+        // save the file
+        // also save flash start and ram start for comparing easily later
+        let mut compatible_tbfs: Vec<(&TbfFile, u64, u64)> = Vec::new();
+        for file in &self.tbf_files {
+            let (arch, flash, ram) = Self::split_arch(file.filename.to_string());
+            // check if we have the same arch
+            // check if flash and ram fit
+            if flash != 0 && ram != 0 {
+                if arch.starts_with(settings.arch.as_ref().unwrap())
+                    && flash >= settings.start_address
+                    && ram >= settings.ram_start_address
+                {
+                    compatible_tbfs.push((file, flash, ram));
+                }
+            } else if arch.starts_with(settings.arch.as_ref().unwrap()) {
+                // this happens for C apps, we'll have
+                // arch = "cortex-m4.tbf"
+                // without any flash and ram values
+                compatible_tbfs.push((file, flash, ram));
+            }
+        }
+        Ok(compatible_tbfs)
+    }
+
+    fn split_arch(filename: String) -> (String, u64, u64) {
```

Comment:
```suggestion
    /// ....
    struct TbfFilename {
        /// <put doc comment here>
        pub arch: String,
        /// ...
        pub flash_address: Option<u64>,
        /// ...
        pub ram_address: Option<u64>,
    }
    
    impl TbfFilename {
        /// ...
        pub fn new(...) -> Self { ... }
        
        /// ... (this gets you the original file name in the archive)
        pub fn to_string(&self) -> String {...}
    }

    fn parse_tbf_filename(filename: String) -> TbfFilename {
```

The type is quite complex already, let's just make it a struct

Reply:
and also renamed the function